packages/vira-ci-types/src/Vira/CI/Pipeline/Type.hs

```diff
@@ -39,8 +46,36 @@ newtype SignoffStage = SignoffStage
   }
   deriving stock (Generic, Show)
 
-makeFieldLabelsNoPrefix ''ViraPipeline
-makeFieldLabelsNoPrefix ''BuildStage
-makeFieldLabelsNoPrefix ''AtticStage
-makeFieldLabelsNoPrefix ''CachixStage
-makeFieldLabelsNoPrefix ''SignoffStage
+-- HasField instances for enabling OverloadedRecordUpdate syntax
+
+-- BuildStage field instances
+instance HasField "enable" BuildStage Bool where
+  hasField (BuildStage enable overrideInputs) = ((`BuildStage` overrideInputs), enable)
+
+instance HasField "overrideInputs" BuildStage [(Text, Text)] where
+  hasField (BuildStage enable overrideInputs) = (BuildStage enable, overrideInputs)
+
+-- AtticStage field instances
+instance HasField "enable" AtticStage Bool where
+  hasField (AtticStage enable) = (AtticStage, enable)
+
+-- CachixStage field instances
+instance HasField "enable" CachixStage Bool where
+  hasField (CachixStage enable) = (CachixStage, enable)
+
+-- SignoffStage field instances
+instance HasField "enable" SignoffStage Bool where
+  hasField (SignoffStage enable) = (SignoffStage, enable)
+
+-- ViraPipeline stage instances
+instance HasField "build" ViraPipeline BuildStage where
+  hasField (ViraPipeline build attic cachix signoff) = (\x -> ViraPipeline x attic cachix signoff, build)
+
+instance HasField "attic" ViraPipeline AtticStage where
+  hasField (ViraPipeline build attic cachix signoff) = (\x -> ViraPipeline build x cachix signoff, attic)
+
+instance HasField "cachix" ViraPipeline CachixStage where
+  hasField (ViraPipeline build attic cachix signoff) = (\x -> ViraPipeline build attic x signoff, cachix)
+
+instance HasField "signoff" ViraPipeline SignoffStage where
+  hasField (ViraPipeline build attic cachix signoff) = (ViraPipeline build attic cachix, signoff)
```

Comment:
We could potentially automate this using generics-sop or something. https://www.reddit.com/r/haskell/comments/1n5a170/comment/nfnpq74/